### ISSUE TITLE
Don't post preview comment until actual GitHub Pages deployment workflow completed?

### ISSUE THREAD

**Issue by vincerubinetti** [rossjrw/pr-preview-action#25]
Testing this out, and it works great! One thing I noticed is that, understandably, the action only waits for itself to finish to post the comment on the pull request. It would be great if that could somehow be delayed until the actual GitHub Pages deployment process has completed so that the preview link works as soon as it's posted. Right now, there's a short period of time between when the comment is posted and when the link will actually go somewhere.

I'm using this action in a website template that will be used by many non-technical users, and I could see this confusing them.

As for how to implement this, I'm not sure. Is it possible for this Action to somehow add the associated GitHub Pages deployment run as a Status Check to the PR? Maybe it's possible to use [action-wait-for-check](https://github.com/fountainhead/action-wait-for-check) or some similar action to wait for the deployment to succeed in the middle of running this action (I guess after committing to the `gh-pages` branch so the build starts, but before returning successfully).

Or maybe a stop gap could simply be adding a message to the pull request comment that clarifies that the preview WILL be available (after a short moment) at the following link. Maybe you could a [GitHub Actions status badge](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge#using-the-workflow-file-name) in the comment, which hopefully should update immediately to say "building"?

**Comment by rossjrw** [comment]
Good idea! Off the top of my head, I don't think this action can get any information about the Pages build process itself. This is definitely worth spending some time looking into - thanks for the contribution!

**Comment by vincerubinetti** [comment]
Yeah, there doesn't seem to be a way to programmatically get info about the process at all.

Since users of this action can't customize the text in the pr comment, I'd strongly request tweaking the wording carefully so people inexperienced with GitHub won't get confused.

**Comment by ElioDiNino** [comment]
> Since users of this action can't customize the text in the pr comment, I'd strongly request tweaking the wording carefully so people inexperienced with GitHub won't get confused.

I strongly agree here and I really like the addition you added that says "once GitHub [finishes deploying it](https://github.com/vincerubinetti/test/actions/workflows/pages/pages-build-deployment/badge.svg)". I think this would reduce a lot of confusion

**Comment by vincerubinetti** [comment]
I think this endpoint could be used:

https://docs.github.com/en/rest/pages/pages?apiVersion=2022-11-28#get-the-status-of-a-github-pages-deployment

You can pass this argument:

> `pages_deployment_id` Required
The ID of the Pages deployment. You can also give the **commit SHA of the deployment**.

**Comment by rossjrw** [comment]
I've added a 'when github finishes deploying it'-like text to the default comment in [v1.6.0](https://github.com/rossjrw/pr-preview-action/releases/tag/v1.6.0) (#97). I'll be honest, the reason I let your suggestion languish for so long was because I didn't think it was really necessary, even though it was obviously and a good idea and you even cited real user experience; until I had that exact same moment of thinking the action was broken while working on 97.

<img src="https://raw.githubusercontent.com/rossjrw/pr-preview-action/refs/heads/main/.github/sample-preview-link.png">

Not going to close this issue because I'd still like to (have an option to?) have the action wait for the deployment to actually finish before it posts its comment.

Suggestion for myself: current comment text before deployment has finished, wait for deployment, edit comment to indicate that it finished. If comment:false, don't post either comment, but still delay action completion until the deployment is done.

**Comment by rossjrw** [comment]
Merged and released in [v1.7.0](https://github.com/rossjrw/pr-preview-action/releases/tag/v1.7.0) with new input param `wait-for-pages-deployment` (bool, default false). Thanks for the contribution @vincerubinetti!

cc @ElioDiNino as you commented, just in case you are still using this action

**Comment by ElioDiNino** [comment]
> Merged and released in [v1.7.0](https://github.com/rossjrw/pr-preview-action/releases/tag/v1.7.0) with new input param `wait-for-pages-deployment` (bool, default false). Thanks for the contribution @vincerubinetti!
> 
> cc @ElioDiNino as you commented, just in case you are still using this action

Sweet thanks, will try it out!

**Comment by ElioDiNino** [comment]
@rossjrw The new option works largely as expected but the hardcoded 60 second timeout has caused some issues for me since GitHub Pages sometimes takes a little longer than that to finish the build and deploy. Could this be configurable and/or increased to 2-3 minutes instead?

**Comment by rossjrw** [comment]
@ElioDiNino thanks for providing those logs, that's very helpful - in 1.7.2 I'll drop the build start timeout back down to 90s but raise the build finish timeout to 1800s. I figure once the build has started it doesn't really matter how long it takes - that should avoid 99.9% of situations like this. the 180s apiece in 1.7.1 should cover most use cases for now, including ri-zhan's
